scatter/gather reporter transfer (#26954)
idaholab/moose: framework/doc/content/source/transfers/MultiAppReporterTransfer.md

the same way on the parallel processes. Each source app is then matched to the target app with the same
subapp index.

## Distributed Vector Transfer

The MultiAppReporterTransfer also supports transferring reporter vectors in a distributed fashion using the [!param](/Transfers/MultiAppReporterTransfer/distribute_reporter_vector) parameter.

> In this mode, the transfer assumes a one-to-many or many-to-one relationship

Author: I was trying to say that setting that param to true would perform the distributed action. Whether it was one-to-many or many-to-one would depend on the setting of the transfer. Whether you are sending to sub apps or receiving.

In this mode, the transfer assumes a one-to-many or many-to-one relationship
between the reporter values in the main application and the sub-applications.
The reporter value in the main application is expected to be a vector with a
size matching the number of sub-applications. Each sub-application will
send/receive its respective component of the vector based on its index. The main
app reporter is assumed to be replicated while the subapp reporters are assumed
to be root or replicated.

| Main App | Sub App 1 | Sub App 2 | Sub App 3 | Sub App 4 |
|---|---|---|---|---|
| $\begin{pmatrix}1\\2\\3\\4\end{pmatrix}$ | 1 | 2 | 3 | 4 |
|
| $\begin{pmatrix}1\\2\end{pmatrix}$ $\begin{pmatrix}3\\4\\5\end{pmatrix}$ $\begin{pmatrix}6\\7\\8\\9\end{pmatrix}$ $\begin{pmatrix}10\\11\\12\end{pmatrix}$ | $\begin{pmatrix}1\\2\end{pmatrix}$ | $\begin{pmatrix}3\\4\\5\end{pmatrix}$ | $\begin{pmatrix}6\\7\\8\\9\end{pmatrix}$ | $\begin{pmatrix}10\\11\\12\end{pmatrix}$ |

!alert Caution title=[!param](/Transfers/MultiAppReporterTransfer/distribute_reporter_vector) only works with certain reporter modes.
The main
app reporter is assumed to be replicated while the subapp reporters are assumed

> to be root or replicated. Using distributed reporter modes might not give correct answers.

Reviewer: let s error and say that instead of 'might be wrong'

the same way on the parallel processes. Each source app is then matched to the target app with the same
subapp index.

## Distributed Vector Transfer

The MultiAppReporterTransfer also supports transferring reporter vectors in a distributed fashion using the [!param](/Transfers/MultiAppReporterTransfer/distribute_reporter_vector) parameter.
In this mode, the transfer assumes a one-to-many or many-to-one relationship
between the reporter values in the main application and the sub-applications.
The reporter value in the main application is expected to be a vector with a
size matching the number of sub-applications. Each sub-application will
send/receive its respective component of the vector based on its index. The main
app reporter is assumed to be replicated while the subapp reporters are assumed
to be root or replicated.

> | Main App | Sub App 1 | Sub App 2 | Sub App 3 | Sub App 4 |

Reviewer: ```suggestion
!table caption=....
| Main App | Sub App 1 | Sub App 2 | Sub App 3 | Sub App 4 |
```

|---|---|---|---|---|
| $\begin{pmatrix}1\\2\\3\\4\end{pmatrix}$ | 1 | 2 | 3 | 4 |
|
| $\begin{pmatrix}1\\2\end{pmatrix}$ $\begin{pmatrix}3\\4\\5\end{pmatrix}$ $\begin{pmatrix}6\\7\\8\\9\end{pmatrix}$ $\begin{pmatrix}10\\11\\12\end{pmatrix}$ | $\begin{pmatrix}1\\2\end{pmatrix}$ | $\begin{pmatrix}3\\4\\5\end{pmatrix}$ | $\begin{pmatrix}6\\7\\8\\9\end{pmatrix}$ | $\begin{pmatrix}10\\11\\12\end{pmatrix}$ |

> !alert warning title=[!param](/Transfers/MultiAppReporterTransfer/distribute_reporter_vector)  is only implemented for certain reporter modes.

Reviewer: use caption instead of warning

Author: Did you mean "note"?

!listing reporter_transfer/main.i block=int_to_int int_from_int string_from_string string_to_string caption=Main application reporter transfers
   indent=2 header=[Transfers] footer=[]

### Distributed Reporter

Here we are transferring a vector reporter and a vector of vectors reporter in a

> scatter and gather fashion.

Reviewer: say which app has the vector of vecotrs

!listing reporter_transfer/main.i block=int_to_int int_from_int string_from_string string_to_string caption=Main application reporter transfers
   indent=2 header=[Transfers] footer=[]

### Distributed Reporter

Here we are transferring a vector reporter and a vector of vectors reporter in a
scatter and gather fashion. The main application holds a vector of vectors and a

> single vector,while each subapp has a vector and a Real (scalar) reporter.

Reviewer: switch what the subapp has ? and add (respectively)

> single vector,while each subapp has a vector and a Real (scalar) reporter.

Author: I was confused here. What do you mean by "switch what the subapp has ? and add (respectively)"?

This test shows the
scatter operation by distributing a vector and a single Real to each subapp and
shows the gather operation by aggregating the same values into a

> different set of reporters. This is to demonstrate that order of the

Reviewer: it would be clearer to have two inputs, one for scatter and one for gather

!listing reporter_transfer/main.i block=int_to_int int_from_int string_from_string string_to_string caption=Main application reporter transfers
   indent=2 header=[Transfers] footer=[]

### Distribution of a vector Reporter

#### Scatter vector Reporter

Here we are transferring a vector reporter and a vector of vectors reporter in a
scatter fashion. The main application holds a vector of vectors and a
single vector, while each subapp has a vector and a Real (scalar) reporter.
During the transfer the subapp's reporters are populated from the scattering of

> the mainapp's vector reporter and a vector of vectors reporter.

Reviewer: ```suggestion
the main app's vector reporter. The main app can also use a vector of vectors reporter, in which case those vectors
are distributed to the subapps.
```

!listing dist_vector/main_send.i block=Reporters caption=Main application reporters

!listing dist_vector/sub.i block=Reporters caption=Sub-application reporters

!listing dist_vector/main_send.i block=Transfers caption=Main application reporter transfers
   indent=2 header=[Transfers] footer=[]

#### Gather vector Reporter

> Here we are transferring a vector reporter and a vector of vectors reporter in a gather fashion. The main application holds a vector of vectors and a single vector, while each subapp has a vector and a Real (scalar) reporter. This test shows the gather operation by aggregating the same values from each subapp into a set of reporters in the main application.

Reviewer: why a set of report?
isnt it a vector report?

Reviewer: ```suggestion
Here we are transferring a vector reporter (and as a second example, a vector of vectors reporter) in a gather fashion. The main application holds a vector of vectors and a single vector, while each subapp has a vector and a Real (scalar) reporter. This test shows the gather operation by aggregating the same values from each subapp into a set of reporters in the main application.
```

Author: I used the word set loosely. Fix it to be more precise.
